### PR TITLE
Issue 19: Update langkit version

### LINKED ISSUE #19: Update langkit version
An update of langkit is needed to fix Componolit/RecordFlux#603. The issue is fixed on langkit's master branch (https://github.com/AdaCore/langkit/commit/02991cc2b0696b371cf9ead6297d0ed2913f3f28).

### PR DESCRIPTION
**Pull request by senier**
# Description

This updates the langkit version to the current stable commit. It also links all Ada libraries statically into the resulting shared parser library and removes the `patchelf` hack. Tested with GNAT Community 2020 and Pro 21.0.

Closes #19

# Review Checklist

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [x] Can this solution be simplified?
- [x] Does this change add unwanted compile-time or run-time dependencies?
- [x] Was a framework, API, library, service used that should not be used?
- [x] Was a framework, API, library, service not used that could improve the solution?
- [x] Is the code at the right abstraction level?
- [x] Is the code modular enough?
- [x] Would you have solved the problem in a different way that is substantially better in terms of the code’s maintainability, readability, performance, security?
- [x] Does similar functionality already exist in the codebase? If so, why isn’t this functionality reused?
- [x] Are there any best practices, design patterns or language-specific patterns that could substantially improve this code?

### Logic Errors and Bugs

- [x] Can you think of any use case in which the code does not behave as intended?
- [x] Can you think of any inputs or external events that could break the code?

### Error Handling and Logging

- [x] Is error handling done the correct way?
- [x] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?
- [x] Are there enough log events and are they written in a way that allows for easy debugging?
- [x] Is the style of error / logging messages consistent?

### Dependencies

- [x] If this change requires updates outside of the code, like updating the documentation, configuration, readme files, was this done?
- [x] Might this change have any ramifications for other parts of the system, or backward compatibility?

### Security and Data Privacy

- [x] Does this code open the software up for security vulnerabilities?
- [x] Does this code change reveal some secret information like keys, passwords, or usernames (e.g., checked-in credentials)?

### Performance

- [x] Do you think this code change will impact system performance in a negative way?
- [x] Do you see any potential to improve the performance of the code?

### Usability

- [x] Is the proposed solution well designed from a usability perspective?
- [x] Is the API well documented?
- [x] Is the API/UI intuitive to use?

### Testing

- [x] Does it have enough automated tests (unit/integration/system tests)?
- [x] Are there some test cases, input or edge cases that should be tested in addition?

### Readability

- [x] Was the code easy to understand?
- [x] Can the readability of the code be improved by smaller methods?
- [x] Can the readability of the code be improved by different function/method or variable names?
- [x] Is the code located in the right file/folder/package?
- [x] Do you think certain methods should be restructured to have a more intuitive control flow?
- [x] Is the data flow understandable?
- [x] Are there redundant comments?
- [x] Could some comments convey the message better?
- [x] Would additional comments make the code more understandable?
- [x] Could some comments be removed by making the code itself more readable?
- [x] Is there any commented out code?

### Experts Opinion

- [x] Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be committed?
- [x] Will this code change impact different teams?
- [x] Should they have a say on the change as well?

### CODE REVIEW

**Review by treiher — CHANGES_REQUESTED** [review]

File: setup.py [hunk 1]
```diff
@@ -1,33 +1,19 @@
...
-# "$ORIGIN" to the RPATH of librflxlang.so. The simplest way to
-# do this is the `patchelf` tool.
-def patch_rpath() -> None:
-    sopath = "lib/librflxlang/relocatable/dev/librflxlang.so"
-    rpath = subprocess.check_output(["patchelf", "--print-rpath", sopath])
-    subprocess.run(
-        ["patchelf", "--set-rpath", "$ORIGIN:" + rpath.decode("utf-8"), sopath], check=True
-    )
+class InstallError(BaseException):
+    pass
```
[the remark this answers is not in the document]
Comment (senier) [reply setup.py]: Fixed in 59500989cbaf3d71e1531821463cedad6ed3fab6

File: setup.py [hunk 2]
```diff
@@ -121,8 +158,7 @@ def run(self) -> None:
         "Topic :: System :: Networking",
```
[the remark this answers is not in the document]
Comment (senier) [reply setup.py:160]: Done in 3e49e92. This reminded me of the other new upstream feature that we planned to use. I now set the version for the generated parser and we should be able to retrieve it in RecordFlux. This also closes #6.

**Review by treiher — APPROVED** [review]
